### PR TITLE
Test 'Where does my donation go' link

### PR DESCRIPTION
**Pull request by Sperling-0**

### CODE REVIEW

File: config/banners.js [hunk 1]
```diff
@@ -39,6 +39,9 @@ export const bannerConfig = {
 			mini: '.wmde-banner-mini-button',
 			form: '.wmde-banner-form-button'
 		},
+		where_does_my_donation_go_link: {
```
Comment (Abban) [inline config/banners.js]: It'd probably be better to call this `use_of_funds` (as that's what we call the content in the modal) and the sub item isn't a form but a link. Also you can add a selector to check if the modal is open:

```suggestion
		use_of_funds: {
		    link: '.wmde-banner-footer-usage-link',
		    modal_is_open: '.banner_modal.is-visible'
```

File: pages/Banner.js [hunk 2]
```diff
@@ -43,6 +43,11 @@ export default class Banner {
 		return this;
 	}
 
+	clickWhereDoesMyDonationGoLink() {
+		this.dandy.click( bannerConfig.selectors.where_does_my_donation_go_link.form );
```
Comment (Abban) [inline pages/Banner.js]:
```suggestion
		this.dandy.click( bannerConfig.selectors.use_of_funds.link );
```

File: pages/Banner.js [hunk 3]
```diff
@@ -43,6 +43,11 @@ export default class Banner {
 		return this;
 	}
 
+	clickWhereDoesMyDonationGoLink() {
```
Comment (Abban) [inline pages/Banner.js]:
```suggestion
	clickUseOfFundsLink() {
```

File: tests/banners/where-does-my-donation-go.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { bannerConfig } from '../../config/banners.js';
```
Comment (Abban) [inline tests/banners/where-does-my-donation-go.js]: This file should be called something more descriptive of the action, maybe `desktop-shows-use-of-funds.js`

File: pages/Banner.js [hunk 5]
```diff
@@ -43,6 +43,11 @@ export default class Banner {
...
+		this.dandy.click( bannerConfig.selectors.where_does_my_donation_go_link.form );
+		return this;
+	}
+
```
Comment (Abban) [inline pages/Banner.js:63]: You need to add a method that checks that the Use of Funds is visible:

```js
checkUseOfFundsIsVisible() {
	this.dandy.checkElementExists( bannerConfig.selectors.use_of_funds.modal_is_open );
	return this;
}
```

File: tests/banners/where-does-my-donation-go.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import buildBannerTestConfig from '../../src/build_banner_test_config.js';
+
+const testConfig = buildBannerTestConfig( process.argv );
+
+( async () => {
+
+	const banner = new Banner( testConfig.url, bannerConfig.selectors, testConfig.parameters, testConfig.options );
+
+	await banner.waitForBanner()
+		.clickWhereDoesMyDonationGoLink()
```
Comment (Abban) [inline tests/banners/where-does-my-donation-go.js]:
```suggestion
		.clickUseOfFundsLink()
		.checkUseOfFundsIsVisible()
```

**Review by Abban — CHANGES_REQUESTED** [review]
I forgot to add to the new documentation that each test should check for something as that's what these tests are for: `Perform actions -> Check result is as expected`. I've mentioned what you can do to add that in the review comments.

The screenshots aren't used for the tests other than letting you see what the page looks like if a test fails. They're more of a developer tool, you'll see the real feedback in your terminal when you run the tests.

This is a good start, though!

**Review by gbirke — APPROVED** [review]

File: pages/Banner.js [hunk 7]
```diff
@@ -43,6 +43,23 @@ export default class Banner {
 		return this;
 	}
 
+	clickUseOfFundsLink() {
+		this.dandy.click( bannerConfig.selectors.use_of_funds.link );
+		return this;
+	}
+
+	checkUseOfFundsIsVisible() {
+		this.dandy.checkElementExists( bannerConfig.selectors.use_of_funds.modal_is_open );
```
Comment (Abban) [inline pages/Banner.js:52]: All you need to check here is if the modal is open as that's what this test is for. You can leave the selectors in the config though, as we might write a test that needs to select them in the future.

**Comment by Abban** [comment]
This was done in https://github.com/wmde/Dandy/pull/8

**Pull request closed without merging**